src/python/pants/backend/jvm/targets/jvm_target.py

```diff
@@ -74,6 +75,9 @@ def __init__(self,
                               rollbacks, but in certain cases may conflict with user libraries.
     :type zinc_file_manager: bool
     """
+    deprecated_conditional(lambda: resources is not None, '1.5.0dev0',
```

Comment:
Either way, we should clean this up uniformly in a separate change (I had already pushed this one).